hw/xbox/xbox.c

```diff
@@ -250,31 +250,40 @@ uint8_t *load_eeprom(void)
             return NULL;
         }
 
-        fd = open(filename, O_RDONLY | O_BINARY);
-        if (fd < 0) {
+        eeprom_fileDescriptor = open(filename, O_RDWR | O_BINARY);
+        if (eeprom_fileDescriptor < 0) {
             fprintf(stderr, "qemu: EEPROM file '%s' could not be opened.\n", filename);
             g_free(filename);
             exit(1);
             return NULL;
         }
 
-        rc = read(fd, eeprom_data, eeprom_size);
+        rc = read(eeprom_fileDescriptor, eeprom_data, eeprom_size);
         if (rc != eeprom_size) {
             fprintf(stderr, "qemu: Could not read the full EEPROM file.\n");
-            close(fd);
+            close(eeprom_fileDescriptor);
             g_free(filename);
             exit(1);
             return NULL;
         }
 
-        close(fd);
         g_free(filename);
     } else {
         memcpy(eeprom_data, default_eeprom, eeprom_size);
     }
     return eeprom_data;
 }
 
+void save_eeprom(uint8_t *eepromBuffer)
+{
+   int status = -1;
+
+    if(eeprom_fileDescriptor > 0)  {
+        lseek(eeprom_fileDescriptor,0,SEEK_SET);
+        status = write(eeprom_fileDescriptor, eepromBuffer, 256);
```

Comment:
What if this fails?

Reply:
We most likely already printed an error (while reading the eeprom file)
So if we can't write to the file we don't try to :)